sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -288,9 +288,10 @@ case class TruncateTableCommand(
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
     if (!catalog.tableExists(tableName)) {
-      logError(s"table '$tableName' in TRUNCATE TABLE does not exist.")
+      throw new AnalysisException(s"Table '$tableName' in TRUNCATE TABLE does not exist.")
```

Comment:
This is my bad. It seemed a bit harsh to throw an exception in this case. `DropTableCommand` does a similar thing.